lib/meals/requests.js

```diff
@@ -25,14 +25,19 @@ export const addFoodToMeal = (food, meal) => {
   let calories = food.getElementsByTagName('p')[1].innerText
   let data = { name: name, calories: calories }
   var mealId = addMeal.getMealId(meal);
+  let foodAttributes = food.getElementsByTagName('p')
+  let foodObject = {name: foodAttributes[0].innerText, calories: parseInt(foodAttributes[1].innerText)}
   return fetch(APIhost + `/api/v1/meals/` + mealId + `/foods/` + foodId, {
     method: 'post',
     headers:
       { 'Content-Type': 'application/json' },
     body: JSON.stringify(data)
   })
     .then(response => response.json())
-    // .then(getMeals)
+    .then(addMeal.prependToMeal(foodObject, {name: meal}))
+    .then(updateDiary.updateRemainingCaloriesPostCreate(foodObject, meal))
```

Comment:
Let's pull this out in one method- or at least couple the total table's update

Reply:
yup good idea. I'll check into that now!